bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java

```diff
@@ -153,27 +160,44 @@ private void createReminderText(final EMFDataBindingContext dbc, final Composite
             reminder.setText(Messages.reminderForStepMessage);
             autoGeneratedOperationButton.setText(Messages.autoGeneratedOperationStepButton);
             manuallyDefinedOperationButton.setText(Messages.manuallyDefinedOperationStepButton);
-            dbc.bindValue(actionObservable, PojoObservables.observeValue(this, "isAutogeneratedOperation"));
+            binding = dbc.bindValue(actionObservable, PojoObservables.observeValue(this, "isAutogeneratedOperation"));
         } else {
             reminder.setText(Messages.reminderForProcessMessage);
             autoGeneratedOperationButton.setText(Messages.autoGeneratedOperationProcessButton);
             manuallyDefinedOperationButton.setText(Messages.manuallyDefinedOperationProcessButton);
-            dbc.bindValue(actionObservable, PojoObservables.observeValue(this, "isAutogeneratedScript"));
+            binding = dbc.bindValue(actionObservable, PojoObservables.observeValue(this, "isAutogeneratedScript"));
         }
     }
 
+    public void checkMannuallyDefinedOperationButton() {
+        if (contract.eContainer() instanceof Task) {
+            isAutogeneratedOperation = false;
+        } else {
+            isAutogeneratedScript = false;
+        }
+        binding.updateModelToTarget();
+    }
+
     /**
      * @return the isAutogeneratedOperation
      */
     public boolean getIsAutogeneratedOperation() {
-        return isAutogeneratedOperation;
+        if (contract.eContainer() instanceof OperationContainer) {
+            return isAutogeneratedOperation;
+        } else {
+            return false;
+        }
     }
 
     /**
      * @return the isAutogeneratedScript
      */
     public boolean getIsAutogeneratedScript() {
```

Comment:
if you do that you can't use PojoObservables.observeValue, it doesn't work, i tried before nad it was the only way to make it work